src/Worker/Core/Shims/TaskOrchestrationContextWrapper.cs

```diff
@@ -179,9 +179,8 @@ public override async Task<TResult> CallSubOrchestratorAsync<TResult>(
         static string? GetInstanceId(TaskOptions? options)
             => options is SubOrchestrationOptions derived ? derived.InstanceId : null;
         string instanceId = GetInstanceId(options) ?? this.NewGuid().ToString("N");
-        string defaultVersion = this.invocationContext.Options?.Versioning?.DefaultVersion ?? string.Empty;
-        string version = options is SubOrchestrationOptions subOptions ? subOptions.Version : defaultVersion;
-
+        string defaultVersion = this.GetDefaultVersion();
+        string version = options is SubOrchestrationOptions { Version: { } v } ? v.Version : defaultVersion;
```

Comment:
I don't understand this C# syntax, but I'll take your word for it that it's good and correct.